pkg/imageverifycache/client.go

```diff
@@ -67,27 +78,41 @@ func WithTTLDuration(t time.Duration) Option {
 	}
 }
 
+func generateKey(policy kyvernov1.PolicyInterface, ruleName string, imageRef string) string {
+	return string(policy.GetUID()) + ";" + policy.GetResourceVersion() + ";" + ruleName + ";" + imageRef
+}
+
 func (c *cache) Set(ctx context.Context, policy kyvernov1.PolicyInterface, ruleName string, imageRef string) (bool, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	c.logger.Info("Setting cache", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
+	c.logger.WithValues("Setting cache", "namespace", policy.GetNamespace(), "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
 	if !c.isCacheEnabled {
 		return false, nil
 	}
-	c.logger.Info("Successfully set cache", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
+	key := generateKey(policy, ruleName, imageRef)
+
+	stored := c.cache.SetWithTTL(key, nil, 1, c.ttl)
+	if stored {
+		c.logger.WithValues("Successfully set cache", "namespace", policy.GetNamespace(), "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
+		return true, nil
+	}
 	return false, nil
 }
 
 func (c *cache) Get(ctx context.Context, policy kyvernov1.PolicyInterface, ruleName string, imageRef string) (bool, error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
-
-	c.logger.Info("Searching in cache", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
+	c.logger.WithValues("Searching in cache", "namespace", policy.GetNamespace(), "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
 	if !c.isCacheEnabled {
 		return false, nil
 	}
-	c.logger.Info("Cache entry not found", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
-	c.logger.Info("Cache entry found", "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
+	key := generateKey(policy, ruleName, imageRef)
+	_, found := c.cache.Get(key)
+	if found {
+		c.logger.WithValues("Cache entry found", "namespace", policy.GetNamespace(), policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
+		return true, nil
+	}
+	c.logger.WithValues("Cache entry not found", "namespace", policy.GetNamespace(), "policy", policy.GetName(), "ruleName", ruleName, "imageRef", imageRef)
```

Comment:
You don't want to capitalize log messages.

And should we move these logs to the higher level (v=4) and leave the error logs at v=2?

Reply:
done